### PR TITLE
Add tab width option

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This adds a new `tab_width` option that gets passed to the Printer. The `tab_width` option specifies the visual width of a tab character. 

## Test Plan

I added a new test. The fact that I had to add options not to regress tests that assumed a tab width of 8 proves that it works :)

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6814** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6814" target="_blank">Graphite</a> 
    * **PR #6815** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6815" target="_blank">Graphite</a> 
      * **PR #6816** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6816" target="_blank">Graphite</a> 
        * **PR #6817** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6817" target="_blank">Graphite</a> 
          * **PR #6848** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6848" target="_blank">Graphite</a>  👈
          * **PR #6819** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6819" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/6848?utm_source=stack-comment).

**Review by konstin — COMMENTED** [review]
How does this interact with docstring formatting where we have "pad to the next multiple of 8"  hardcoded?

File: crates/ruff_formatter/src/lib.rs [hunk 1]
```diff
@@ -108,6 +108,35 @@ impl std::fmt::Display for IndentStyle {
...
+    pub const fn value(&self) -> u32 {
+        self.0.get() as u32
+    }
+}
+
+impl Default for TabWidth {
+    fn default() -> Self {
+        // SAFETY: 2 is guaranteed to not be 0
+        #[allow(unsafe_code)]
+        Self(unsafe { NonZeroU8::new_unchecked(2) })
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply crates/ruff_formatter/src/lib.rs:128]: The constness is unstable but yeah, this is probably not worth it

**Comment by MichaReiser** [comment]
> How does this interact with docstring formatting where we have "pad to the next multiple of 8" hardcoded?

Thanks for pointing this out. I changed it to use the default from the options

**Comment by MichaReiser** [comment]
Merging because it is a simple enough change and there's other work that depends on it.

**Pull request merged**